src/client/js/settings.js

```diff
@@ -0,0 +1,160 @@
+    }
+  })
+}
+
+const settingsClose = document.getElementById('js-settings-close')
+if (settingsClose) {
+  settingsClose.addEventListener('click', async _ => {
+    try {
+      const addEmailDialog = document.getElementById('js-settings-modal')
+      addEmailDialog.close()
+    } catch (err) {
+      console.error(`Error: ${err}`)
+    }
+  })
+}
+
+const settingsAddVerification = document.getElementById('js-settings-modal-send-verification')
+if (settingsAddVerification) {
+  settingsAddVerification.addEventListener('click', async _ => {
+    const email = document.getElementById('js-settings-email-modal-input').value
+    const csrfToken = document
+      .getElementById('js-settings')
+      .getAttribute('data-csrf-token')
+
+    const response = await fetch('/api/v1/user/email', {
+      headers: {
+        'Content-Type': 'application/json',
+        'x-csrf-token': csrfToken
+      },
+      mode: 'same-origin',
+      method: 'POST',
+      body: JSON.stringify({ email })
+    })
+
+    if (!response?.ok) {
+      throw new Error(`Sending verification email failed: ${response?.error}`)
+    }
+
+    const addEmailDialogContent = document.getElementById('js-settings-modal-content')
+    // TODO: Localize string below
+    addEmailDialogContent.textContent = `Verify the link sent to ${email} to add it to Firefox Monitor. Manage all email addresses in Settings.`
+
+    const addEmailDialogControls = document.getElementById('js-settings-modal-controls')
+    addEmailDialogControls.hidden = true
+
+    const addEmailDialog = document.getElementById('js-settings-modal')
+    addEmailDialog.addEventListener('close', _ => {
+      addEmailDialogControls.hidden = false
+      return window.location.reload(true)
+    }, { once: true })
+  })
+}
+
+const settingsRemoveEmailButtons = document.getElementsByClassName('js-remove-email-button')
+if (settingsRemoveEmailButtons?.length) {
+  for (const removeEmailButton of settingsRemoveEmailButtons) {
+    removeEmailButton.addEventListener('click', async event => {
+      try {
+        const subscriberId = event.target.getAttribute('data-subscriber-id')
+        const emailId = event.target.getAttribute('data-email-id')
+        const csrfToken = document
+          .getElementById('js-settings')
+          .getAttribute('data-csrf-token')
+
+        const response = await fetch('/api/v1/user/remove-email', {
+          headers: {
+            'Content-Type': 'application/json',
+            'x-csrf-token': csrfToken
+          },
+          mode: 'same-origin',
+          method: 'POST',
+          body: JSON.stringify({ emailId, subscriberId })
+        })
+
+        if (response && response.redirected === true) {
+          return window.location.reload(true)
+        }
+      } catch (err) {
+        console.error(`Error: ${err}`)
+      }
+    })
+  }
+}
+
+const settingsResendEmailLinks = document.getElementsByClassName('js-settings-resend-email')
+if (settingsResendEmailLinks?.length) {
```

Comment:
Minor note, non-blocking: instead of using a conditional to check each interactive element on the page, you could simply verify that this is the settings partial.  Each "main" template is created with the partial name as a data attribute, so you could do:
```js
const settingsPartial = document.querySelector("[data-partial='settings']")
```
followed by a single conditional check similar to:
```js
if (settingsPartial) init()
```
Where `init()` is a function to set up listeners, run-once code, etc.

In addition to feeling confident the correct partial and elements are loaded, you have access to the `<main>` wrapper around your partial structure, and can query within it (`settingsPartial.getElementsByClassName('js-settings-resend-email'`) instead of the entire document.  This might allow less verbose names?

If interested, see https://github.com/mozilla/blurts-server/blob/main/src/client/js/partials/breaches.js for example.

Reply:
I like that suggestion — it also helps to grasp which dynamic elements are expected to be present easily.